### ISSUE TITLE
no unicode support

### ISSUE THREAD

**Issue by msva** [laanwj/bitcoin-qt#14]
Hi there! I've some wallets named in unicode, and when I open them in subj — i see broken encoded strings. Can you fix it? ;)

**Comment by laanwj** [comment]
I see -- unicode for communication with the bitcoin core seems to be broken :(

**Comment by msva** [comment]
i've almost fix it ;)
But i can't did last commit because my notebook is broken :'(
I'll try to commit it in few days to my fork and export your changes and then ask you to pull, ok?
Btw, i've added few mare translations there! ;)

**Comment by laanwj** [comment]
My idea would be to add two functions to GUIUtil

```
// To std::string. Works like QString::toStdString but understands
```

unicode.
    static std::string toStdString(const QString &str);

```
// From std::string. Works like QString::fromStdString but understands
```

unicode.
    static QString fromStdString(const std::string &str);

Then replace all usages of fromStdString and toStdString throughout the
program with these.

**Comment by msva** [comment]
I've done it in that way:
I change all
**bla.ToStdString()**
with labels (and all, that is not an "address" (because address _**SHOULD**_ be ascii)) to
**(std::string)bla.toLocal8bit().data()**
and it seems, that it working. But I think, that my way is so spiked... Your way is quite better ;)

In other hand - i've already add many new tr()'s and _()'s in code.
Let me try to commit last changes from broken note and after that you can export and review my changes, okay? :)

**Comment by laanwj** [comment]
OK, I'll wait for your pull request, we can refactor after that if needed

**Comment by laanwj** [comment]
Any luck with this?

**Comment by msva** [comment]
Partialy :'(
I've almost finish it, but I can't fix QString::fromStdString() to something more unicode-friendly...
At the moment it is broken encoding in splash, in wallet history, and in all places, where "QString::fromStdString" is using to get info from upstream part.
May be you can help with it? :)

**Comment by laanwj** [comment]
To do correct utf-8 conversion, GuiUtil::toStdString(str) should do something like

```
QByteArray arr = str.toUtf8();
return std::string(arr.data(), arr.size());
```

GuiUtil::fromStdString(str) should do 

```
return QString::fromUtf8(str.data(), str.size());
```

I'll take a further look at it after I nail wallet encryption

**Comment by laanwj** [comment]
I think we can do it even simpler; QString::fromStdString and QString::toStdString use the CString codec. If this is defined as  UTF-8, the problem is solved without using different functions (essentially they do the same as above). Cannot try right now, but it should be this  (once, at the beginning of the program):

```
QTextCodec::setCodecForCStrings(QTextCodec::codecForName("UTF-8"));
```

Strings with non-ASCII characters will then survive the QString->std::string->QString translation :)

**Comment by laanwj** [comment]
Seems to work... with the setting, taking an unicode character, converting it to std::string and back preserves it:

```
#include <QString>
#include <QTextCodec>
#include <string>
#include <QDebug>

int main()
{
    QTextCodec::setCodecForCStrings(QTextCodec::codecForName("UTF-8"));

    QString test(QChar(0x26b8));
    std::string test2 = test.toStdString();
    QString test3 = QString::fromStdString(test2);
    printf("%04x %s\n", test.at(0).unicode(), test2.data());
    if(test == test3)
    {
        printf("OK\n");
    }
    else
    {
        printf("fail\n");
    }
}
```

Outputs "26b8 ⚸ OK".

**Comment by msva** [comment]
ok, i've added it and translated almost all strings.
Now i've only problem, that it is still english in options dialog and in payment info, but I think, it can be fixed later.
Now you can check out my fork and merge it in main tree ;)

**Comment by msva** [comment]
btw, i've updated master branch there. Please check, that you're merging **7****ea6e6135884aed405d4cb597a488492cd4105ce** commit ;)

**Comment by msva** [comment]
Closed due to #20 (pull req)
